cyences_app_for_splunk/bin/device_inventory_gen.py

```diff
@@ -0,0 +1,445 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import csv
+import uuid
+import time
+import json
+from datetime import datetime
+
+from splunklib.searchcommands import dispatch, EventingCommand, Configuration, Option
+from splunklib.searchcommands.validators import Validator
+from splunk import rest
+import cs_utils
+
+
+import logging
+import logger_manager
+logger = logger_manager.setup_logging('device_inventory_command', logging.DEBUG)
+
+
+LOOKUP_DIR = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'lookups')
+
+def get_lookup_path(lookup_name):
+    return os.path.join(LOOKUP_DIR, lookup_name)
+
+
+# DEVICE_INVENTORY_LOOKUP = get_lookup_path('cs_device_inventory_lookup.csv')
+DEVICE_INVENTORY_LOOKUP_COLLECTION = 'cs_device_inventory_collection'
+DEVICE_INVENTORY_LOOKUP_HEADERS = ['uuid', 'time', 'ip', 'hostname', 'mac_address', 'tenable_uuid', 'qualys_id', 'lansweeper_id', 'sophos_uuid', 'crowdstrike_userid', 'windows_defender_host']
+# DEVICE_INVENTORY_LOOKUP_HEADERS_KEY_INDEX = {'uuid':0, 'time':1, 'ip':2, 'hostname':3, 'mac_address':4, 'tenable_uuid':5, 'qualys_id':6, 'lansweeper_id':7, 'sophos_uuid':8, 'crowdstrike_userid':9, 'windows_defender_host':10}
+# LOOKUP_KEY_UUID = 0
+# LOOKUP_KEY_TIME = 1
+# LOOKUP_KEY_IP = 2
+# LOOKUP_KEY_HOSTNAME = 3
+# LOOKUP_KEY_MAC_ADDRESS = 4
+DEVICE_INVENTORY_LOOKUP_HEADERS_KEY_INDEX = {'uuid':'uuid', 'time':'time', 'ip':'ip', 'hostname':'hostname', 'mac_address':'mac_address', 'tenable_uuid':'tenable_uuid', 'qualys_id':'qualys_id', 'lansweeper_id':'lansweeper_id', 'sophos_uuid':'sophos_uuid', 'crowdstrike_userid':'crowdstrike_userid', 'windows_defender_host':'windows_defender_host'}
+LOOKUP_KEY_UUID = 'uuid'
+LOOKUP_KEY_TIME = 'time'
+LOOKUP_KEY_IP = 'ip'
+LOOKUP_KEY_HOSTNAME = 'hostname'
+LOOKUP_KEY_MAC_ADDRESS = 'mac_address'
+
+DEVICE_INVENTORY_LOOKUP_BACKUP_PREFIX = 'backup_device_inventory'
+
+
```

Comment:
Instead of removing functions of CSV files which is used for debugging, try to add VARIABLE which will indicate whether we are in debug mode or not. If it is in debug mode then only csv files should be created.

Reply:
Added code for that.